pydra/engine/task.py

```diff
@@ -288,7 +288,20 @@ def __init__(
 
     @property
     def command_args(self):
-        """Get command line arguments."""
+        """Get command line arguments, returns a list if task has a state"""
+        if is_lazy(self.inputs):
+            raise Exception("can't return cmdline, self.inputs has LazyFields")
+        if self.state:
+            command_args_list = []
+            self.state.prepare_states(self.inputs)
```

Comment:
what happens if you call this multiple times? will it still work?

Reply:
it works. I agree that it's not the best solution, and should be review where `prepare_States` is called